src/components/Tabs/Tab.tsx

```diff
@@ -0,0 +1,176 @@
+import styled from '@emotion/styled'
+import {
+  ComponentProps,
+  ElementType,
+  ForwardedRef,
+  KeyboardEventHandler,
+  MouseEventHandler,
+  ReactNode,
+  Ref,
+  forwardRef,
+  useMemo,
+} from 'react'
+import Tooltip from '../Tooltip'
+import { useTabsContext } from './TabsContext'
+
+const StyledCounter = styled.span`
+  border-radius: 50%;
+  font-size: 12px;
+  background-color: ${({ theme }) => theme.colors.neutral.backgroundStrong};
+  border: 1px solid ${({ theme }) => theme.colors.neutral.borderStrong};
+  margin-left: ${({ theme }) => theme.space['1']};
+  height: 24px;
+  width: 24px;
+  text-overflow: ellipsis;
+  line-height: 22px;
+`
+
+const BadgeContainer = styled.span`
+  margin-left: ${({ theme }) => theme.space['1']};
+  display: flex;
+`
+
+export const StyledTabButton = styled.button<{
+  'data-variant'?: 'menuitem' | 'tab'
+}>`
+  display: flex;
+  flex-direction: row;
+  padding: ${({ theme }) => `${theme.space['1']} ${theme.space['2']}`};
+  cursor: pointer;
+  justify-content: center;
+  align-items: center;
+  font-size: 16px;
+  line-height: 24px;
+  white-space: nowrap;
+  color: ${({ theme }) => theme.colors.neutral.textWeak};
+  font-weight: 500;
+  text-decoration: none;
+  user-select: none;
+  touch-action: manipulation;
+  transition: color 0.2s;
+  border: none;
+  background: none;
+  border-bottom-width: 2px;
+  border-bottom-style: solid;
+  border-bottom-color: ${({ theme }) => theme.colors.neutral.borderWeak};
+  outline: none;
```

Comment:
Here also we have a lot of props linked to text maybe we should use a component text inside of this component?